docs/migrating-from-0.9.md

```diff
@@ -0,0 +1,251 @@
+
+## Beta APIs
+
+We're introducing an unstable, Beta API channel, and leaving session, chat and Genkit client APIs in beta as we continue to refine them. 
+
+Note: When using the APIs as part of the Beta API, you may experience breaking changes outside of SemVer. Breaking chanhes may occur on minor release (we'll try to avoid making breaking beta API changes on patch releases).
+
+**Old:**
+
+```ts
+import { genkit } from 'genkit';
+const ai = genkit({...})
+const session = ai.createSession({ ... })
+```
+
+**New**
+
+```ts
+import { genkit } from 'genkit/beta';
+const ai = genkit({...})
+const session = ai.createSession({ ... })
+```
+
+**Old:**
+
+```ts
+import { runFlow, streamFlow } from 'genkit/client';
+```
+
+**New**
+
+```ts
+import { runFlow, streamFlow } from 'genkit/beta/client';
+```
+
+## Introducing new `@genkit-ai/express` package
+
+The package contains utilities to make it easier to build an express app using Express.js.
+
+Refer to https://js.api.genkit.dev/modules/_genkit-ai_express.html for more details.
+
+`startFlowServer` has moved from part of the genkit object to the express package, to use startFlowServer, update your imports. 
+
+
+**Old:**
+
+```ts
+const ai = genkit({ ... });
+ai.startFlowServer({
+  flows: [myFlow1, myFlow2],
+});
+```
+
+**New**
+
+```ts
+import { startFlowServer } from '@genkit-ai/express';
+startFlowServer({
+  flows: [myFlow1, myFlow2],
+});
+```
+
+## Changes to Flows
+
+`run` function for custom trace blocks has moving to `ai.run`
+
+**Old:**
+
+```ts
+ai.defineFlow({name: 'banana'}, async (input, streamingCallback) => {
+  const step = await run('myCode', async () => {
+    return 'something'
+  });
+})
+```ts
+
+**New**
+
+```ts
+ai.defineFlow({name: 'banana'}, async (input, {context, sendChunk}) => {
+  const step = await ai.run('myCode', async () => {
+    return 'something'
+  });
+})
+```
+
+`ai.defineStreamingFlow` has been removed. Use `ai.defineFlow` instead. Also, `streamingCallback` moved to a field inside the second argument of he flow function and is now called `sendChunk`.
+
+**Old:**
+
+```ts
+const flow = ai.defineStreamingFlow({name: 'banana'}, async (input, streamingCallback) => {
+  streamingCallback({chunk: 1});
+})
+
+const {stream} = await flow()
+for await (const chunk of stream) {
+  // ...
+}
+```
+
+**New**
+
+```ts
+const flow = ai.defineFlow({name: 'banana'}, async (input, {context, sendChunk}) => {
+  sendChunk({chunk: 1});
+})
+
+const {stream, output} = flow.stream(input);
+for await (const chunk of stream) {
+  // ...
+}
+```
+
+Flow auth is now called context. You can access auth as a field inside context:
+
+**Old:**
+
+```ts
+ai.defineFlow({name: 'banana'}, async (input) => {
+  const auth = getFlowAuth();
+  // ...
+})
+```
+
+**New**
+
+```ts
+ai.defineFlow({name: 'banana'}, async (input, { context }) => {
+  const auth = context.auth;
+})
+```
+
+`onFlow` moved to `firebase-functions/https` package and got renamed to `onCallGenkit`. Here's how to use it:
```

Comment:
This would benefit from an `old` version to show how this is different than the code would have been before

Reply:
done.